src/Sulu/Component/Content/Import/Webspace.php

```diff
@@ -0,0 +1,511 @@
+        BasePageDocument $document,
+        $structureType,
+        $webspaceKey,
+        $locale,
+        $format,
+        $data
+    ) {
+        $structure = $this->structureManager->getStructure($structureType);
+        $properties = $structure->getProperties(true);
+        $node = $this->documentRegistry->getNodeForDocument($document);
+        $node->setProperty(sprintf('i18n:%s-template', $locale), $structureType);
+        $state = $this->getParser($format)->getPropertyData('state', $data, null, null, 2);
+        $node->setProperty(sprintf('i18n:%s-state', $locale), $state);
+
+        if ($this->getParser($format)->getPropertyData('title', $data) === '') {
+            throw new \Exception(
+                sprintf('Document(%s) has not set any title', $document->getUuid())
+            );
+
+            return;
+        }
+
+        // import all content data
+        foreach ($properties as $property) {
+            $value = $this->getParser($format)->getPropertyData(
+                $property->getName(),
+                $data,
+                $property->getContentTypeName()
+            );
+
+            // don't generate a new url when one exists
+            $doImport = true;
+            if ($property->getContentTypeName() == 'resource_locator') {
+                if (!$document->getResourceSegment()) {
+                    $parent = $document->getParent();
+
+                    if ($parent instanceof BasePageDocument) {
+                        $parentUuid = $parent->getUuid();
+                        $value = $this->generateUrl(
+                            $structure->getPropertiesByTagName('sulu.rlp.part'),
+                            $parentUuid,
+                            $webspaceKey,
+                            $locale,
+                            $format,
+                            $data
+                        );
+                    }
+                } else {
+                    $doImport = false;
+                }
+            }
+
+            // import property data
+            if ($doImport) {
+                $this->importProperty($property, $node, $structure, $value, $webspaceKey, $locale, $format);
+            }
+        }
+
+        // import extensions
+        $extensions = $this->extensionManager->getExtensions($structureType);
+
+        foreach ($extensions as $key => $extension) {
+            $this->importExtension($extension, $key, $node, $data, $webspaceKey, $locale, $format);
+        }
+
+        // set required data
+        $document->setTitle($this->getParser($format)->getPropertyData('title', $data));
+    }
+
+    /**
+     * @param BasePageDocument $document
+     * @param string $structureType
+     * @param string $webspaceKey
+     * @param string $locale
+     * @param string $format
+     * @param array $data
+     */
+    protected function setDocumentSettings($document, $structureType, $webspaceKey, $locale, $format, $data, $overrideSettings)
```

Comment:
does this fit into a single line? or should it be splitted in multiple lines? us type-hint for document